core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -199,6 +210,25 @@ protected void doStop() throws Exception {
     ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
     context.setContextPath("/");
     context.addServlet(servletHolder, "/*");
+    String authMethod = getConfiguration().getString(RestConfig.AUTHENTICATION_METHOD);
+    if (!"NONE".equals(authMethod)) {
+      String realm = getConfiguration().getString(RestConfig.AUTHENTICATION_REALM);
+      List<String> roles = getConfiguration().getList(RestConfig.AUTHENTICATION_ROLES);
+      final ConstraintSecurityHandler securityHandler = new ConstraintSecurityHandler();
+      Constraint constraint = new Constraint();
+      constraint.setAuthenticate(true);
+      constraint.setRoles(roles.toArray(new String[roles.size()]));
+      ConstraintMapping constraintMapping = new ConstraintMapping();
+      constraintMapping.setConstraint(constraint);
+      constraintMapping.setMethod("*");
+      constraintMapping.setPathSpec("/*");
+      securityHandler.addConstraintMapping(constraintMapping);
```

Comment:
@norwood If we can figure out how to shoehorn in XML configs as well, I think that's fine. There's certainly a ton of example configs out there that users would benefit from if we could handle arbitrary Jetty configs.